### PR TITLE
Create the markPullRequestsAsDeployed action

### LINKED ISSUE #1575: [GH Actions] Create the markPullRequestsAsDeployed action
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

# Problem
There is no obvious audit trail on Expensify.cash pull requests that show us whether they've been deployed.

# Why is this important?
Right now, this isn't a super critical issue because PRs are always deployed when they're merged. But soon, that won't be the case. Furthermore, right now we're only deploying once, directly to production, but soon we'll be deploying pull requests first to staging, then to production.

# Solution
Create a Github Action called `markPullRequestsAsDeployed` that takes the following inputs:

1) `PR_LIST`: `Array<Number>` – The list of PR #’s to comment in
2) `IS_PRODUCTION_DEPLOY`: `Boolean`

And it will use [`octokit.issues.createComment`](https://octokit.github.io/rest.js/v18#issues-create-comment) to comment on every PR in the list stating that it was deployed to staging or production.

## To be considered complete

- You must provide documentation that this has been tested in a separate repo (or a fork of Expensify.cash) as is working as expected.
- Just because the PR is merged doesn't mean that your work is done. We need to make sure that it's working as expected with a live test in Expensify.cash.

[Internal issue](https://github.com/Expensify/Expensify/issues/155147)
[Upwork posting](https://www.upwork.com/jobs/~0178ef13cafb2ddedd)

### PR DESCRIPTION
**Pull request by attaradev**
### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
Fixes #1575 

### Tests
Tested the action on another repo, and the comments were written based on the given inputs

### Screenshots
Screenshot 2021-03-05 at 12 32 48 AM
Screenshot 2021-03-05 at 12 33 06 AM
Screenshot 2021-03-05 at 12 33 49 AM

### CODE REVIEW

**Comment by mikeattara** [comment]
I have read the CLA Document and I hereby sign the CLA

File: .github/actions/markPullRequestsAsDeployed/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: 'Mark Pull Requests as Deployed'
+description: 'Mark pull requests as deployed on production or staging'
+inputs:
+  PR_LIST:
+    description: 'Array or pull request numbers'
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/action.yml:5]:
```suggestion
    description: 'Array of pull request numbers'
```

File: .github/actions/markPullRequestsAsDeployed/action.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    required: true
+    default: '[]'
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/action.yml:7]: Seems weird to give a required parameter a default.
```suggestion
```

File: .github/actions/markPullRequestsAsDeployed/action.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  IS_PRODUCTION_DEPLOY:
+    description: 'Check if deploying to production'
+    required: true
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/action.yml:10]:
```suggestion
    required: false
```

File: .github/actions/markPullRequestsAsDeployed/action.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    default: 'false'
+  GITHUB_TOKEN:
+    description: 'GitHub Token'
+    required: true
+    default: "${{ github.token }}"
+runs:
+  using: 'node12'
+  main: 'dist/index.js'
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/action.yml:18]: This seems wrong, I think it needs to be:

```suggestion
  main: './index.js'
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+/**
+ * Create comment on pull request
+ *
+ * @param {Number} number - The pull request number
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:7]:
```suggestion
 * @param {Number} number - The pull request or issue number
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {String} messageBody - The comment message
+ * @param {Object} octokitClient - The ocktokit client
+ * @returns {Promise}
+ */
+async function createComment(number, messageBody, octokitClient) {
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:12]: Sadly Expensify doesn't use `async/await` in any of our Javascript code. There is a good explanation of this in our [style guide](https://github.com/Expensify/Expensify.cash/blob/master/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not). So you'll need to convert all your code to promises instead. Sorry 🙃

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        });
+        return response;
+    } catch (error) {
+        console.log(`Unable to write comment on #${number}`);
+        core.setFailed(error);
+    }
+}
+
+try {
+    const prList = core.getInput('PR_LIST', {required: true});
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:28]:
```suggestion
    const prList = JSON.parse(core.getInput('PR_LIST', {required: true}));
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    const isProd = core.getInput('IS_PRODUCTION_DEPLOY', {required: true});
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:29]:
```suggestion
    const isProd = JSON.parse(core.getInput('IS_PRODUCTION_DEPLOY', {required: true}));
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    const token = core.getInput('GITHUB_TOKEN', {required: true});
+    const IS_PRODUCTION_DEPLOY = JSON.parse(isProd);
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:31]:
```suggestion
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    const PR_LIST = JSON.parse(prList);
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:32]:
```suggestion
```

File: .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    const DATE = new Date();
```
Comment (roryabraham) [inline .github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js:33]:
```suggestion
    const date = new Date();
```

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by mikeattara** [comment]
I have made the changes

**Comment by mikeattara** [comment]
I am closing this PR so that I can fix the file changes

**Pull request closed without merging**